src/resolve/resolved-dns-transaction.c

```diff
@@ -1697,10 +1698,16 @@ static int dns_transaction_prepare(DnsTransaction *t, usec_t ts) {
                 /* Let's then prune all outdated entries */
                 dns_cache_prune(&t->scope->cache);
 
+                /* For the initial attempt or when no stale data is requested, disable serve stale and answer the question from the cache (honors ttl property).
+                 * On the second attempt, if StaleRetentionSec is greater than zero, try to answer the question using stale date (honors until property) */
+                uint64_t query_flags = t->query_flags;
+                if (t->n_attempts == 1 || t->scope->manager->stale_retention_usec == 0)
+                        SET_FLAG(query_flags, SD_RESOLVED_NO_STALE, true);
```

Comment:
just do:

```c
query_flags |= SD_RESOLVED_NO_STALE;
```

use the SET_FLAG() macro only if you actually have a boolean variable you want to propagate into a flag, but here you don't you just want to enable the flag